assets/src/design-system/components/typeahead/typeahead.js

```diff
@@ -0,0 +1,350 @@
+ * @param {number} props.popupFillWidth Allows for an override of how much of popup width to take up for dropDown.
+ * @param {number} props.popupZIndex Allows for an override of the default popup z index (2).
+ * @param {Function} props.renderItem If present when menu is open, will override the base list items rendered for each option, the entire item and whether it is selected will be returned and allow you to style list items internal to a list item without affecting dropdown functionality.
+ * @param {string} props.selectedValue The selected value, should correspond to a value in the options array of objects.
+ *
+ */
+
+export const Typeahead = ({
+  ariaInputLabel = DEFAULT_ARIA_INPUT_LABEL,
+  ariaClearLabel = DEFAULT_ARIA_CLEAR_LABEL,
+  disabled,
+  handleTypeaheadValueChange,
+  hasError,
+  hint,
+  isFlexibleValue,
+  isKeepMenuOpenOnSelection,
+  onMenuItemClick,
+  options = [],
+  placeholder = DEFAULT_PLACEHOLDER,
+  placement = PLACEMENT.BOTTOM,
+  popupFillWidth = DEFAULT_POPUP_FILL_WIDTH,
+  popupZIndex,
+  selectedValue = '',
+  ...rest
+}) => {
+  const listId = useMemo(() => `list-${uuidv4()}`, []);
+  const inputId = useMemo(() => `select-button-${uuidv4()}`, []);
+  const clearId = useMemo(() => `clear-button-${uuidv4()}`, []);
+
+  const inputRef = useRef();
+  const menuRef = useRef();
+
+  const {
+    activeOption,
+    inputValue,
+    isMenuFocused,
+    isOpen,
+    normalizedOptions,
+  } = useTypeahead({
+    isFlexibleValue,
+    options,
+    selectedValue,
+    handleTypeaheadValueChange,
+  });
+
+  /**
+   * Callbacks that begin typeahead interaction
+   * handleInputFocus: When input is given focus, set menu to isOpen and reset isMenuFocused to false.
+   * handleInputClick: When input is clicked, toggle menu.
+   * */
+  const handleInputFocus = useCallback(() => {
+    isOpen.set(true);
+    isMenuFocused.set(false);
+  }, [isOpen, isMenuFocused]);
+
+  const handleInputClick = useCallback(
+    (event) => {
+      event.preventDefault();
+      isOpen.set((prevIsOpen) => !prevIsOpen);
+    },
+    [isOpen]
+  );
+
+  /**
+   * Callbacks passed to menu
+   * handleDismissMenu: when menu is focused out (if isFlexibleValue is false (default), set input back to selectedValue.)
+   * handleMenuClick: When item menu is chosen by either cursor or keyboard.
+   */
+
+  const handleDismissMenu = useCallback(
+    (event) => {
+      // don't dismiss menu if clicking on clear button while menu is open
+      if (
+        clearId &&
+        (event?.srcElement?.id === clearId ||
+          event?.srcElement?.nearestViewportElement?.id === clearId)
+      ) {
+        return;
+      }
+
+      if (!isFlexibleValue && activeOption?.label) {
+        inputValue.set(activeOption.label);
+      }
+
+      isOpen.set(false);
+      isMenuFocused.set(false);
+
+      // move focus to next element if available.
+      inputRef.current?.offsetParent?.nextSibling?.focus();
```

Comment:
🤔 do we need this? If this element doesn't trap the focus I'm guessing we can just rely on the browser to find the next available element right?